### PR TITLE
fix to avoid common hook warning if no hook is overridden

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixed a small mistake made here: https://github.com/PyTorchLightning/pytorch-lightning/pull/10603
which is raising the warning always on master even when the hook is not overridden

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @justusschock @awaelchli @ninginthecloud

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -672,6 +676,15 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed passing `_ddp_params_and_buffers_to_ignore` ([#11949](https://github.com/PyTorchLightning/pytorch-lightning/pull/11949))
 
 
+- Fixed an `AttributeError` when calling `save_hyperparameters` and no parameters need saving ([#11827](https://github.com/PyTorchLightning/pytorch-lightning/pull/11827))
```
Comment (rohitgr7) [inline CHANGELOG.md:679]: the other entries were incorrectly moved to the already released section somewhere in other PRs. Verified that they are not present on release branch, so updating them here.

**Review by carmocca — APPROVED** [review]

File: tests/trainer/connectors/test_data_connector.py [hunk 2]
```diff
@@ -123,7 +124,6 @@ def test_override_both_model_and_datamodule(self, hook_name):
         with pytest.warns(UserWarning, match=f"have overridden `{hook_name}` in both"):
             hook = trainer._data_connector._datahook_selector.get_hook(hook_name)
 
-        warning_cache.clear()
```
Comment (krshrimali) [inline tests/trainer/connectors/test_data_connector.py:126]: Hmm, do you know why was this present before? And if you could share why this is removed?

**Review by krshrimali — CHANGES_REQUESTED** [review]
Just a small question.

File: tests/trainer/connectors/test_data_connector.py (hunk 2 above)
Comment (rohitgr7) [reply tests/trainer/connectors/test_data_connector.py:126]: it's not removed. If have moved up so that it will be refreshed with each test.

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by krshrimali — APPROVED** [review]
LGTM!

**Pull request merged**